src/main.rs

```diff
@@ -22,6 +24,43 @@ mod dvi;
 mod framebuffer;
 mod link;
 
+global_asm! {
+    "
+    .section .text
+    .align 4
+    data_cpy_table:
+     .word _scratch_x_source
+     .word _scratch_x_start
+     .word _scratch_x_end
+     .word _scratch_y_source
+     .word _scratch_y_start
+     .word _scratch_y_end
+     .word 0
+    .global __pre_init
+    .type __pre_init,%function
+    .thumb_func
+    __pre_init:
+     push {{r4, lr}}
+     ldr r4, =data_cpy_table
+
+    1:
```

Comment:
Where are the labels `1` and `2` referenced?.

Reply:
These are [local labels](https://developer.arm.com/documentation/dui0068/b/Assembler-Reference/Symbols/Local-labels). You can read "b 1f" and "b 2b" as "branch 1 forward" and "branch 2 backward." I was also initially thrown off, but they're quite common in ARM asm and they're growing on me.